### PR TITLE
expression: implement vectorized evaluation for builtinLastDaySig

### PR DESCRIPTION
**Pull request by ekalinin**
PCP #12101

### What problem does this PR solve? 

This PR implements vectorized `builtinLastDaySig`

### What is changed and how it works?

```bash
➜ make vectorized-bench VB_FILE=Time VB_FUNC=builtinLastDaySig                           
cd ./expression && \
	go test -v -benchmem \
		-bench=BenchmarkVectorizedBuiltinTimeFunc \
		-run=BenchmarkVectorizedBuiltinTimeFunc \
		-args "builtinLastDaySig"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-12    	1000000000	         0.0728 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinLastDaySig-VecBuiltinFunc-12         	   66322	     18120 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinLastDaySig-NonVecBuiltinFunc-12      	   23734	     50426 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	4.308s
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -697,11 +697,33 @@ func (b *builtinGetFormatSig) getFormat(format, location string) string {
...
+		return err
+	}
+	times := result.Times()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if times[i].InvalidZero() {
+			return handleInvalidTimeError(b.ctx,
+				types.ErrIncorrectDatetimeValue.GenWithStackByArgs(times[i].String()))
```
Comment (qw4990) [inline expression/builtin_time_vec.go]: If this error is handled here, we should continue to calculate the next row:
```
If err := handleInvalidTimeError(...); err != nil {
    return err
}
result.SetNull(i, true)
continue
```

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -697,11 +697,33 @@ func (b *builtinGetFormatSig) getFormat(format, location string) string {
...
+		}
+		tm := times[i].Time
+		year, month := tm.Year(), tm.Month()
```
Comment (SunRunAway) [inline expression/builtin_time_vec.go]: A curiosity is that why this judgment is deleted?
https://github.com/pingcap/tidb/blob/6b9703bc5f8c77ce3350c8880958b44fef53b7b7/expression/builtin_time.go#L6536-L6538

File: expression/builtin_time_vec.go (hunk 1 above)
Comment (SunRunAway) [reply expression/builtin_time_vec.go]: Do we need this judgment?
I do not see it in builtinLastDaySig.evalTime

File: expression/builtin_time_vec.go (hunk 2 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: It was not deleted. It's the same as `times[i].InvalidZero()`

File: expression/builtin_time_vec.go (hunk 1 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: Fixed

File: expression/builtin_time_vec.go (hunk 2 above)
Comment (SunRunAway) [reply expression/builtin_time_vec.go]: That's great, could you also modify the original implementation `builtinLastDaySig.evalTime`.

Comment (ekalinin) [reply expression/builtin_time_vec.go]: Done

**Comment by ekalinin** [comment]
Why it was closed?

**Comment by ekalinin** [comment]
@SunRunAway @qw4990 could you reopen?

**Pull request merged**